packages/web-config-server/src/__tests__/apiV1/dataBuilders/helpers/calculateOperationForAnalytics.test.js

```diff
@@ -0,0 +1,324 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { arrayToAnalytics } from '@tupaia/data-broker';
+import { NO_DATA_AVAILABLE } from '/apiV1/dataBuilders/constants';
+
+import { calculateOperationForAnalytics } from '/apiV1/dataBuilders/helpers';
+
+const models = {};
+
+const analytics = arrayToAnalytics([
+  ['temperature', 'TO', '20220101', 2],
+  ['temperature', 'TO', '20220101', 5],
+  ['temperature', 'TO', '20220101', 7],
+  ['temperature', 'TO', '20220101', 2],
+
+  ['result', 'TO', '20220101', 'Positive'],
+  ['result', 'TO', '20220101', 'Positive'],
+  ['result', 'TO', '20220101', 'Positive Mixed'],
+  ['result', 'TO', '20220101', 'Negative'],
+
+  ['uniqueCode', 'TO', '20220101', 'Yes, more than 100'],
+  ['uniqueCodeForName', 'TO', '20220101', 'Octavia'],
+
+  ['height', 'TO', '20220101', 15],
+  ['width', 'TO', '20220101', 2],
+
+  ['Flower_found_Daisy', 'TO', '20220101', 1],
+  ['Flower_found_Tulip', 'TO', '20220101', 'No'],
+  ['Flower_found_Orchid', 'TO', '20220101', 1],
+  ['Flower_found_Orchid', 'TO', '20220101', 1],
+
+  ['Best_Superhero1', 'TO', '20220101', 'SuperGirl'],
+  ['Best_Superhero2', 'TO', '20220101', 'Black Widow'],
+  ['Best_Superhero3', 'TO', '20220101', 'My Sister'],
+
+  ['population', 'Melbourne', '19990101', 10],
+  ['population', 'Melbourne', '20050101', 100],
+  ['population', 'Sydney', '20100101', 20],
+  ['population', 'Sydney', '20050101', 200],
+]);
+
+describe('calculateOperationForAnalytics', () => {
+  it('throws if the operation is not defined', async () => {
+    await expect(
+      calculateOperationForAnalytics(models, analytics, { operator: 'NOT_AN_OPERATOR' }),
+    ).toBeRejectedWith('Cannot find operator: NOT_AN_OPERATOR');
+    await expect(calculateOperationForAnalytics(models, analytics, {})).toBeRejectedWith(
+      'Cannot find operator: undefined',
+    );
+  });
+
+  // TODO skipping test cases until the tested functionality is fixed
+  describe.skip('CHECK_CONDITION', () => {
```

Comment:
@IgorNadj @rohan-bes skipping because these tests fail. They were failing under `mocha` too but the tests were not awaited properly, so the failure was reflected in the results:

![calculateOperationForAnalytics errors](https://user-images.githubusercontent.com/20692464/192768704-4ebcbdad-b1e0-4aae-8bbc-1bcf38160d52.png)

I had a look at [calculateOperationForAnalytics.js](https://github.com/beyondessential/tupaia/blob/dev/packages/web-config-server/src/apiV1/dataBuilders/helpers/calculateOperationForAnalytics.js#L38), and I think that this is a fix that makes sense and makes all tests pass:


In:

```js
  const filterKeys = config.filterKeys ?? [config.dataElement];
  const filterValueMap = Object.fromEntries(filterKeys.map(key => [key, config[key]]));
```
Change the first line to:

```js
  const filterKeys = config.filterKeys ?? ['dataElement'];
```

You may want to raise a bug to investigate further :slightly_smiling_face:

Reply:
Ah great stuff @kael89! I've raised a [ticket](https://linear.app/bes/issue/RN-676/investigate-skipped-tests-in-web-config-server-for) for it, but happy to skip for now 👍